components/style/properties/properties.mako.rs

```diff
@@ -1091,6 +1098,10 @@ impl PropertyDeclaration {
                     match input.try(|i| CSSWideKeyword::parse(context, i)) {
                         Ok(CSSWideKeyword::InheritKeyword) => {
                             % for sub_property in shorthand.sub_properties:
+                                if properties_seen.get_${property.ident}() {
```

Comment:
If the concern is that we'll set the bit even with an invalid value for the longhand, I guess that's fair, and you can instead of moving all the `set` calls above the match, add another one below, looping through the subproperties.